### PR TITLE
First step in #114: Provide helpers to calculate absence and suppression state.

### PR DESCRIPTION
**Pull request by lgolding**

### CODE REVIEW

File: src/Sarif.Viewer.VisualStudio/Sarif/Run.extensions.cs [hunk 1]
```diff
@@ -136,5 +136,48 @@ public static bool HasErrorLevelToolExecutionNotifications(this Run run)
 
             return false;
         }
+
+        /// <summary>
```
Comment (lgolding) [inline src/Sarif.Viewer.VisualStudio/Sarif/Run.extensions.cs:140]: >/// [](start = 8, length = 3)

Would these helpers be better off in the SDK?

**Review by GabeDeBacker — APPROVED** [review]
:shipit:

File: src/Sarif.Viewer.VisualStudio/Sarif/Result.extensions.cs [hunk 2]
```diff
@@ -2,6 +2,9 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information. 
 
 using System;
```
Comment (michaelcfanning) [inline src/Sarif.Viewer.VisualStudio/Sarif/Result.extensions.cs:4]: >System [](start = 6, length = 6)

Can you please get these files renamed properly to Result.Extensions.cs, etc., in your next change?

File: src/Sarif.Viewer.VisualStudio/Sarif/Run.extensions.cs (hunk 1 above)
Comment (michaelcfanning) [reply src/Sarif.Viewer.VisualStudio/Sarif/Run.extensions.cs:140]: Agreed, nice helpers. Create an SDK issue?

---
In reply to: [525564884](https://github.com/microsoft/sarif-visualstudio-extension/pull/276#discussion_r525564884) [](ancestors = 525564884)

File: src/Sarif.Viewer.VisualStudio/Sarif/Result.extensions.cs (hunk 2 above)
Comment (lgolding) [reply src/Sarif.Viewer.VisualStudio/Sarif/Result.extensions.cs:4]: You're looking for just the case change 'e' => 'E'? Sure.

File: src/Sarif.Viewer.VisualStudio/Sarif/Run.extensions.cs (hunk 1 above)
Comment (lgolding) [reply src/Sarif.Viewer.VisualStudio/Sarif/Run.extensions.cs:140]: [#2157](https://github.com/microsoft/sarif-sdk/issues/2157): Migrate suppressed and absent issue helpers from VS extension to SDK

---
In reply to: [527881875](https://github.com/microsoft/sarif-visualstudio-extension/pull/276#discussion_r527881875) [](ancestors = 527881875,525564884)

**Pull request merged**